include/deal.II/lac/trilinos_tpetra_vector.h

```diff
@@ -1013,33 +1013,32 @@ namespace LinearAlgebra
       ViewType2d vector_2d_local =
         vector->template getLocalView<Kokkos::HostSpace>(
           Tpetra::Access::ReadWrite);
-      ViewType2d vector_2d_nonlocal =
-        nonlocal_vector->template getLocalView<Kokkos::HostSpace>(
-          Tpetra::Access::ReadWrite);
 #  else
       vector->template sync<Kokkos::HostSpace>();
 
       using ViewType2d =
         decltype(vector->template getLocalView<Kokkos::HostSpace>());
       ViewType2d vector_2d_local =
         vector->template getLocalView<Kokkos::HostSpace>();
-      ViewType2d vector_2d_nonlocal =
-        nonlocal_vector->template getLocalView<Kokkos::HostSpace>();
 #  endif
 
       // Having extracted a view into the multivectors above, now also
-      // extract a view into the one vector we actually store.
+      // extract a view into the one vector we actually store. We can
+      // do this right away for the locally owned part. We defer creating
+      // the view into the nonlocal part to when we know that we actually
+      // need it; this also makes sure that we correctly deal with the
+      // case where we do not actually store a nonlocal part.
       using ViewType1d =
         decltype(Kokkos::subview(vector_2d_local, Kokkos::ALL(), 0));
 
       ViewType1d vector_1d_local =
         Kokkos::subview(vector_2d_local, Kokkos::ALL(), 0);
-      ViewType1d vector_1d_nonlocal =
-        Kokkos::subview(vector_2d_nonlocal, Kokkos::ALL(), 0);
```

Comment:
This should really not be expensive. We only assign a couple of pointers. I'm not sure if using `std::optional` is preferable here. Instead, you could just default construct the object here:
```
ViewType1d vector_1d_nonlocal;
```

Reply:
I needed a way to test at the very bottom of the function whether we have written into any of the nonlocal elements. I do that by testing whether the `std::optional` has been created. So I needed a variable at the top level of the function (of some sort). Would you prefer I just put a `bool` at the top level? I think that the `std::optional` is sort of a cleaner representation of what I'm doing because it only requires one, rather than two variables.